src/App.tsx

```diff
@@ -1,61 +1,128 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { ChangeEvent, FormEvent, useState } from 'react';
+import { getUserById } from './utils/getUserById';
+import { getMaxId } from './utils/getMaxId';
+
+const todosWithUser: Todo[] = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: getUserById(todo.userId),
+  };
+});
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUser);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const handleSetTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value.trimStart());
+
+    if (hasTitleError) {
+      setHasTitleError(false);
+    }
+  };
+
+  const handleSetUser = (event: ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(Number(event.target.value));
+
+    if (hasUserError) {
+      setHasUserError(false);
+    }
+  };
+
+  const handleSetTodos = () => {
+    const newTodo = {
+      id: getMaxId(todos),
```

Comment:
The `getMaxId` function should be used to get the maximum ID from the current todos and then increment it by 1 to generate a new unique ID for the new todo. Currently, it seems to be used directly, which might not be correct if `getMaxId` does not already add 1 to the maximum ID.